### PR TITLE
[BC-14] Add estimator controllers

### PR DESCRIPTION
**Pull request by CorrectACHU**
Added features: -Added Estimator controller. Now Estimator can update tickets, add comments to the ticket, delete comments, see ticket details, see all his tickets. Fixed code: -Did some changes in models - now supplement has one to one relation to ticket. -Moved some classes to new location - like DTO classes.

### CODE REVIEW

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/estimator/EstimatorTicketController.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+        Optional<ResponseEntity<String>> checkAnswer = checkCommentEstimatorOwnership(estimator, comment);
+        if (checkAnswer.isPresent()) {return checkAnswer.get();}
+
+        commentRepository.delete(comment);
+        return ResponseEntity.ok("Comment was deleted!");
+    }
+
+
+    //////////////////////////
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/estimator/EstimatorTicketController.java]: please remove unnecessary comment

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/estimator/EstimatorTicketController.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    @GetMapping("/ticket/{ticketId}")
+    public ResponseEntity getTicketDetails(@PathVariable long ticketId) {
+        Person estimator = (Person) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Ticket ticket = ticketRepository.findById(ticketId).orElse(null);
+        Optional<ResponseEntity<String>> checkAnswer = checkTicketEstimatorOwnership(estimator,ticket);
+        if (checkAnswer.isPresent()) {return checkAnswer.get();}
+
+        TicketDto ticketDto = ticketMapper.toDto(ticket);
+        return ResponseEntity.ok(ticketDto);
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/estimator/EstimatorTicketController.java]: this logic is very similar to the one in User Controller. Does this sound reasonable to extract similar logic into a separate service?
You can pass your checker to the service as Function or custom interface **if neaded** in order to make it agnostic to the checks required.

please do similar for other controller methods if possible

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/manager/dto/TicketManagerDto.java [hunk 3]
```diff
@@ -16,70 +16,15 @@
...
 @Builder
+@Data
 public class TicketManagerDto {
     private Long id;
     private Person currentManager;
     private Person currentEstimator;
     private TicketStatus ticketStatus;
     private OtherCharge otherCharge;
     private Set<Contact> otherContacts;
-    private Set<Supplement> supplements;
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/manager/dto/TicketManagerDto.java:26]: [Please ignore] In reality we indeed can have multiple supplements but I'm ok if it woudl be just one

**Review by rmaskalenka — CHANGES_REQUESTED** [review]

**Comment by CorrectACHU** [comment]
No useful comment , I meant useless :)

File: carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/manager/ManagerTicketController.java [hunk 4]
```diff
@@ -96,30 +103,38 @@ public ResponseEntity<String> deleteCommentToTicket(@PathVariable long commentId
...
+//    }
+//    private static Optional<ResponseEntity<String>> checkCommentOwnership(Person manager, Comment comment) {
+//        if (comment == null) {
+//            return Optional.of(ResponseEntity.badRequest().body("Comment does not exist"));
+//        } else if (comment.getCommentOwner().getId() != manager.getId()) {
+//            return Optional.of(new ResponseEntity<>("This comment does not belong to current manager", HttpStatus.FORBIDDEN));
+//        }
+//        return Optional.empty();
+//    }
+
```
Comment (rmaskalenka) [inline carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/controllers/manager/ManagerTicketController.java]: if you want to remove something just remove it. Main branch sohuld not contain such big commented out chunks of code

**Review by rmaskalenka — APPROVED** [review]

**Pull request merged**